src/main/java/org/jboss/pnc/cleaner/archiver/BuildArchiver.java

```diff
@@ -0,0 +1,437 @@
+        logger.info("Found {} unverified builds.", unverifiedBuilds.size());
+        unverifiedBuilds.forEach(build -> archiveBuild(build));
+    }
+
+    @Timed
+    private RemoteCollection<Build> getUnarchivedBuilds() {
+        BuildsFilterParameters buildsFilterParameters = new BuildsFilterParameters();
+        buildsFilterParameters.setRunning(false);
+        List<String> attributes = Collections.singletonList("!" + BUILD_ARCHIVED);
+        try {
+            return buildClient.getAll(buildsFilterParameters, attributes);
+        } catch (RemoteResourceException e) {
+            logger.error("Cannot read remote builds.", e);
+            return RemoteCollection.empty();
+        }
+    }
+
+    @Timed
+    public void archiveBuild(Build build) {
+        long trimLogSize = 0;
+        if (build.getTemporaryBuild() && build.getStatus() != BuildStatus.SUCCESS) {
+            trimLogSize = trimmedLogMaxSize;
+        }
+
+        logger.info("Reading alignment logs");
+        LogParser alignmentLog = new LogParser(trimLogSize);
+        alignmentLog.addLiteralLines(literalErrors);
+        alignmentLog.addRegExpLines(regExpErrors);
+        try (Response response = finalLogs.getFinalLog(build.getId(), "alignment-log")) {
+            alignmentLog.findMatches(response.readEntity(BufferedReader.class));
+        }
+
+        logger.info("Reading build logs");
+        LogParser buildLog = new LogParser(trimLogSize);
+        alignmentLog.addLiteralLines(literalErrors);
+        alignmentLog.addRegExpLines(regExpErrors);
+        try (Response response = finalLogs.getFinalLog(build.getId(), "build-log")) {
+            buildLog.findMatches(response.readEntity(BufferedReader.class));
+        }
+
+        self.archiveBuildRecord(build, alignmentLog, buildLog);
+
+        try {
+            buildClient.addAttribute(build.getId(), BUILD_ARCHIVED, "true");
+        } catch (RemoteResourceException ex) {
+            logger.error("Failed to mark build as archived in PNC", ex);
+        }
+    }
+
+    @Transactional
+    public void archiveBuildRecord(Build build, LogParser buildLog, LogParser alignmentLog) {
+        ArchviedBuildRecord archviedBuildRecord;
+        try {
+            archviedBuildRecord = mapBuild(build);
+        } catch (RemoteResourceException ex) {
+            throw new RuntimeException("Failed to read build info from PNC", ex);
+        }
+
+        String buildType = build.getBuildConfigRevision().getBuildType().toString();
+
+        archviedBuildRecord.autoAlign = !alignmentLog.contains(DMANIPULATION_DISABLE_TRUE)
+                && !alignmentLog.contains(DMANIPULATION_DISABLE_TRUE1);
+        archviedBuildRecord.brewPullActive = alignmentLog.contains(DBREW_PULL_ACTIVE_TRUE);
+        if (buildLog.contains(FRONTEND_MAVEN_PLUGIN)) {
+            buildType = "MVN-WRAPPED-NPM";
+        }
+        archviedBuildRecord.buildType = buildType;
+
+        archviedBuildRecord.trimmedBuildLog = buildLog.getTrimmedLog();
```

Comment:
I would improve this to match the logic of the trigger:
1) only the builds having a final status SYSTEM_ERROR (the one you get from `proccessErrors`) will have their trimmed logs (both build and alignment) stored
2) the builds having a final status `REJECTED_FAILED_DEPENDENCIES` will have their build logs stored entirely as the trimmed log. No trimmed alignment logs will be stored
3) All the builds not falling into 1 and 2 above, will NOT have any trimmed log stored.

Reply:
deciding if the trimmed log are stored is handled around line 89, see the comment there. (I can change the behaviour if you still want)

Reply:
that is fine, we will have more statuses with trimmed logs stored, but only for the temporary builds, so it's perfect as it is. Thanks!